Client/Frontend/Sync/BraveCore/BraveCoreDataImportExportUtility.swift

```diff
@@ -0,0 +1,475 @@
+                            continue
+                        }
+                        break
+                    }
+                    
+                    Bookmark.addFolder(title: chosenFolder, context: .existing(context))
+                    //try context.save()
+                    topLevelFolder = Bookmark.getTopLevelFolders(context).first(where: { $0.isFolder && $0.customTitle == chosenFolder })
+                }
+                
+                parent = topLevelFolder
+            }
+            
+            // Ensure any enclosing folders are present in the model. The bookmark's
+            // enclosing folder structure should be
+            //    path[0] -> path[1] -> ... -> path[path.count - 1]
+            for folderName in bookmark.path ?? [] {
+                if bookmark.inToolbar && parent == nil && folderName == bookmark.path?.first {
+                    //If we're importing directly to the bookmarks bar, skip over the
+                    // folder named "Bookmarks Toolbar" (or any non-Firefox equivalent).
+                    continue
+                }
+                
+                let index = parent?.children?.first(where: { $0.isFolder && $0.customTitle == folderName })
+                if index == nil {
+                    Bookmark.addFolder(title: folderName, parentFolder: parent, context: .existing(context))
+                    //try context.save()
+                    
+                    parent = parent?.children?.filter({ $0.isFolder && $0.customTitle == folderName }).first
+                } else {
+                    parent = index
+                }
+            }
+            
+            foldersAddedTo.insert(parent)
+            if bookmark.isFolder {
+                Bookmark.addFolder(title: bookmark.title, parentFolder: parent, context: .existing(context))
+                //try context.save()
+            } else if let url = bookmark.url {
+                Bookmark.add(url: url, title: bookmark.title, parentFolder: parent, context: .existing(context))
+                //try context.save()
+                
+                let child = parent?.children?.first(where: { !$0.isFolder && $0.url == url.absoluteString && $0.title == bookmark.title })
+                child?.created = bookmark.creationTime
+                child?.lastVisited = nil
+                //try context.save()
+            }
+        }
+    }
+}
+
+extension BraveCoreDataImportExportUtility {
+    private func convertToChromiumFormat(_ bookmark: Bookmark) -> BraveExportedBookmark {
+        // Tail recursion to map children..
+        return BraveExportedBookmark(title: bookmark.isFolder ? bookmark.customTitle ?? "(No Title)" : bookmark.title ?? "(No Title)", id: Int64(bookmark.order), guid: UUID().uuidString.lowercased(), url: URL(string: bookmark.url ?? ""), dateAdded: bookmark.created ?? Date(), dateModified: bookmark.lastVisited ?? Date(), children: bookmark.children?.map({ convertToChromiumFormat($0) }))
```

Comment:
`No Title` strings should be most likely moved to BraveStrings to translate it, unless we want to have it in english for all languages.
Also break this line, it's too long